Feature/181/search toolbar

mistercapxx opened the pull request:
Implemented new component for OffersPage with search toolbar
![Screenshot 2024-08-06 at 12 41 52](https://github.com/user-attachments/assets/fb7a1859-86ef-419d-a48f-efbd5fadcaf0)

 added : interface Offer {
    category:string;
    subject:string;
    name:string;
    description:string;
}
and useEffect filter to test content searching

Yaroslav-Yakovlev commented:
You have problems with linter
![Screenshot_19](https://github.com/user-attachments/assets/7eda2988-ab4f-4152-9e90-b32cdd3594d9)

miller-113 commented:
It must be connected to the backend.

miller-113 reviewed (changes requested):
Eslint issues and connection to backend.

Yaroslav-Yakovlev commented:
You have problems with linter
![Screenshot_33](https://github.com/user-attachments/assets/ea505eeb-de05-4c08-a476-95163c262c2b)

Commit:
only explore

mistercapxx committed:
Merge branch 'develop' into feature/181/search-toolbar

mistercapxx commented:
> You have problems with linter ![Screenshot_33](https://private-user-images.githubusercontent.com/108124402/358168272-ea505eeb-de05-4c08-a476-95163c262c2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NDI2MjcsIm5iZiI6MTcyMzc0MjMyNywicGF0aCI6Ii8xMDgxMjQ0MDIvMzU4MTY4MjcyLWVhNTA1ZWViLWRlMDUtNGMwOC1hNDc2LTk1MTYzYzI2MmMyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxNVQxNzE4NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNGMyMmFiNzM3YTgyMmRiZTZkMDE0YTM1ZmViOTQyYTY1OWM1MjAyYzQzN2I1MjMyMGIzZDcxOGMzZWZlMzVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.j8QRYMC4CeN7iwr1njU1Cbof8lWlJzxPPfTbndQf9FI)

fixed linter errors when removed unnecessary files

mistercapxx commented:
> It must be connected to the backend.

fixed linter issues. Backend connection is coming next in separate issue that i added today and currently working on

nrevutska commented on src/pages/find-offers/Explore.tsx:
```diff
@@ -1,210 +1,140 @@
+    category:string;
+    subject:string;
+    name:string;
+    description:string;
```
this interface should've stayed the way it was

nrevutska commented on src/pages/find-offers/Explore.tsx:
```diff
@@ -1,210 +1,140 @@
+    handleSearch();
+}
+    };
+    
```
these changes do not reflect actual updates, it's just different spacing.

nrevutska commented on src/pages/find-offers/Explore.tsx:
```diff
@@ -1,210 +1,140 @@
+
+        setFilteredOffers(filtered);
+
+    },[category,subject,searchTerm,offers]);
```
bad formatting here

nrevutska commented on src/pages/find-offers/Explore.tsx:
```diff
@@ -1,210 +1,140 @@
+
+    },[category,subject,searchTerm,offers]);
+
+    const handleSearch =() => {
```
bad formatting here

nrevutska reviewed (changes requested):
check your prettier or vs code settings
use 2 spaces, not 4 for tab spacing.
And it would help if you used prettier to do quick formatting before commits
